Kernel/Arch/aarch64/Time/ARMv8Timer.cpp

```diff
@@ -118,9 +126,30 @@ ErrorOr<void> ARMv8TimerDriver::probe(DeviceTree::Device const& device, StringVi
     // Use the EL1 virtual timer, as that timer should should be accessible to us both on device and in a VM.
     auto interrupt_number = TRY(device.get_interrupt_number(to_underlying(DeviceTreeTimerInterruptIndex::EL1Virtual)));
 
-    auto timer = TRY(ARMv8Timer::initialize(interrupt_number));
+    u32 frequency = 0;
+
+    auto clock_frequency_property = device.node().get_property("clock-frequency"sv);
+    if (clock_frequency_property.has_value()) {
+        if (clock_frequency_property->size() != sizeof(u32)) {
+            dmesgln("ARMv8Timer: \"clock-frequency\" property for \"{}\" has invalid size: {}", device.node_name(), clock_frequency_property->size());
+            return EINVAL;
```

Comment:
Should we fall back to `CNTFRQ_EL0::read()` if the size is wrong, or is it better to EINVAL? (not sure)

Reply:
Hm good question. According to the devicetree binding, this property should only be used when the firmware doesn't write the correct value into the CNTFRQ_EL0 register on bootup (in my case it was zero). So I would say, if the devicetree has that property, we likely shouldn't rely on the register value, as it might be wrong.